PyInstaller/building/osx.py

```diff
@@ -143,24 +143,22 @@ def assemble(self):
                            # side effect, the main application doesn't get one
                            # as well, but at startup time the loader will take
                            # care of transforming the process type.
-                           "LSBackgroundOnly": "0",
+                           "LSBackgroundOnly": False,
 
                            }
 
         # Merge info_plist settings from spec file
         if isinstance(self.info_plist, dict) and self.info_plist:
             info_plist_dict.update(self.info_plist)
 
-        info_plist = u"""<?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-<plist version="1.0">
-<dict>"""
-        for k, v in info_plist_dict.items():
-            info_plist += u"<key>%s</key>\n<string>%s</string>\n" % (k, v)
-        info_plist += u"""</dict>
-</plist>"""
-        with codecs.open(os.path.join(self.name, "Contents", "Info.plist"), "w", "utf-8") as f:
-            f.write(info_plist)
+        plist_filename = os.path.join(self.name, "Contents", "Info.plist")
+        try:
+            # python >= 3.4
+            with open(plist_filename, "wb") as plist_fh:
+                plistlib.dump(info_plist_dict, plist_fh)
+        except AttributeError:
```

Comment:
You can drop this -- Python 3.3 is end of life, so I'd rather not put code in to support it.

Reply:
Actually that's for Python 2.7

Reply:
That would be helpful.

Reply:
writePlist seams to accept an optn file, too. Thus I I suggest changing this into
```
with open(plist_filename, "wb") as plist_fh: 
    try:
        plistlib.dump(info_plist_dict, plist_fh)
    except AttributeError:
         # is_py2
        plistlib.writePlist(info_plist_dict, plist_filename)
```

Ths `is_py2` to to find this place when we will remove support for Python 2 in 2020.